File: frontend/src/types/voting.ts
export type VoteChoice = 'yes' | 'no' | 'abstain'

export interface VotingStats {
  total_votes: number
  yes_votes: number
  no_votes: number
  abstain_votes: number
}

export interface UserVotingState {
  has_voted: boolean
  choice?: VoteChoice | null
}

export interface ActiveVotingSession {
  proposal_id: number
  repository_id: number
  session_id: number
  title: string
  summary: string
  starts_at: string
  ends_at: string
  stats: VotingStats
  user_state: UserVotingState
}

File: backend/app/schemas/voting.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class VotingStats(BaseModel):
    total_votes: int
    yes_votes: int
    no_votes: int
    abstain_votes: int


class UserVotingState(BaseModel):
    has_voted: bool
    choice: Optional[str] = None


class ActiveVotingSession(BaseModel):
    proposal_id: int
    repository_id: int
    session_id: int
    title: str
    summary: str
    starts_at: datetime
    ends_at: datetime
    stats: VotingStats
    user_state: UserVotingState

    class Config:
        from_attributes = True
